File: Algorithms/Dynamic Problems/FlodyWarshall.py
#Python3 mplementation of FloydWarshall Algorithm

#the below represents Number of Vertices in the main graph
v=4

"""Define infinity as the large
   enough value. This value will be 
   used for verticles not connected to each other """

INF=99999

#the function solves all pair shortest path via
# FloydWarshall Algorithm

def FloydWarshall(graph):
    
    """
    dist[][] will be the output
    matrix that will finally 
    have the shortest distances
    between every pair of vertices
    """
    dist=list(map(lambda i: list(map(lambda j:j,i)),graph))
    
    for k in range(v):
        
        for i in range(v):
            
            for j in range(v):
                
                dist[i][j]=min(dist[i][j],dist[i][k]+dist[k][j])
                
    printSolution(dist)
    
    
def printSolution(dist):
    
    print ("Following Matrix Shows the shortest distances between every pair of vertices")
    
    for i in range(v):
        
        for j in range(v):
            
            if dist[i][j]==INF:
                
                print( "%7s" % ("INF"),end=" ")
                
            else:
                
                print( "%7d" % (dist[i][j]),end=" ")
                
            if j==v-1:
                
                print()

graph=[[0  ,5  ,INF ,10 ]
      ,[INF,0  ,3  ,INF]
      ,[INF,INF,0  ,1  ]
      ,[INF,INF,INF,0  ]]


FloydWarshall(graph) 

File: Algorithms/Dynamic Problems/flodyWarshall.c
// Floyd Warshall Algorithm
#include<stdio.h>

// Number of vertices in the graph
#define V 4


#define INF 99999


void printSolution(int dist[][V]);


void floydWarshall (int graph[][V])
{

	int dist[V][V], i, j, k;


	for (i = 0; i < V; i++)
		for (j = 0; j < V; j++)
			dist[i][j] = graph[i][j];


	for (k = 0; k < V; k++)
	{
		// Pick all vertices as source one by one
		for (i = 0; i < V; i++)
		{
			
			for (j = 0; j < V; j++)
			{
				
				if (dist[i][k] + dist[k][j] < dist[i][j])
					dist[i][j] = dist[i][k] + dist[k][j];
			}
		}
	}

	// Print the shortest distance matrix
	printSolution(dist);
}

void printSolution(int dist[][V])
{
	printf ("The following matrix shows the shortest distances"
			" between every pair of vertices \n");
	for (int i = 0; i < V; i++)
	{
		for (int j = 0; j < V; j++)
		{
			if (dist[i][j] == INF)
				printf("%7s", "INF");
			else
				printf ("%7d", dist[i][j]);
		}
		printf("\n");
	}
}

int main()
{
	int graph[V][V] = { {0, 5, INF, 10},
						{INF, 0, 3, INF},
						{INF, INF, 0, 1},
						{INF, INF, INF, 0}
					};

	// Print the solution
	floydWarshall(graph);
	return 0;
}


// Output -

/*
      0      5      8      9
    INF      0      3      4
    INF    INF      0      1
    INF    INF    INF      0

*/

// INF is infinte 
// 0 means points are equal